### ISSUE TITLE
"No space left on device" on Kokoro Windows machines

### ISSUE THREAD

**Issue by markdroth** [grpc/grpc#22103]
https://source.cloud.google.com/results/invocations/dbb4d97f-8bcd-42be-9a33-485af04bedc5/targets/github%2Fgrpc%2Ftoplevel_run_tests_invocations%2Frun_tests_c%2B%2B_windows_dbg_native_x64_cmake_vs2017_buildonly/tests

That log is full of "No space left on device" errors.

I'm also seeing a different error that I suspect is actually related:

https://source.cloud.google.com/results/invocations/0b05dd07-ba0c-4ec5-9d25-b8a36a41c605/targets/github%2Fgrpc%2Ftoplevel_run_tests_invocations%2Frun_tests_c%2B%2B_windows_dbg_native_x64_cmake_vs2017_buildonly/tests

The last bit of the log is:

```
  grpc.dir\Debug\tcp_server_posix.obj
  grpc.dir\Debug\tcp_server_utils_posix_common.obj
  grpc.dir\Debug\tcp_server_utils_posix_ifaddrs.obj
  grpc.dir\Debug\tcp_server_utils_posix_noifaddrs.obj
  grpc.dir\Debug\tcp_server_windows.obj
  grpc.dir\Debug\tcp_uv.obj
  grpc.dir\Debug\tcp_windows.obj
  grpc.dir\Debug\time_averaged_stats.obj
Traceback (most recent call last):
  File "C:\Python27\lib\logging\__init__.py", line 882, in emit
    stream.write(fs % msg)
IOError: [Errno 0] Error
Logged from file jobset.py, line 133
2020-02-21 13:44:48,092 FAILED: cmake [ret=1, pid=3528, time=4234.8sec]
2020-02-21 13:44:48,108 FAILED: Some tests failed
```

**Comment by jtattermusch** [comment]
@yashykt FYI  looks like this could be related to the issues you're facing with https://github.com/grpc/grpc/pull/21361/

**Comment by jtattermusch** [comment]
The kokoro workspace uses an empty 50GB volume `projects/google.com:kokoro/global/images/empty-50g-image` so if our build really fills those 50GB up, we should really slim things down.

**Comment by jtattermusch** [comment]
Looks like the problems started with this diff: https://github.com/grpc/grpc/compare/695c844d0fc5ac216a281418fa521130807fa31f...056a3bc884271055154250e1da591663ad240946 (adding absl), the likely explanation is that we were close to the threshold of available disk space before merging the PR and with this PR we crossed it.

windows portability test is now failing on master consistently:
https://source.cloud.google.com/results/invocations/2589d536-68b0-48fe-a538-d91e1155e6f8/targets/github%2Fgrpc%2Ftoplevel_run_tests_invocations%2Frun_tests_c%2B%2B_windows_dbg_native_x64_cmake_vs2017_buildonly/tests
https://source.cloud.google.com/results/invocations/d8a4f0ab-11d2-4d13-964c-206dfbf1cd22/targets/github%2Fgrpc%2Ftoplevel_run_tests_invocations%2Frun_tests_c%2B%2B_windows_dbg_native_x64_cmake_vs2017_buildonly/tests
https://source.cloud.google.com/results/invocations/4d7ab375-1670-4b95-96be-0a75602af0ed/targets/github%2Fgrpc%2Ftoplevel_run_tests_invocations%2Frun_tests_c%2B%2B_windows_dbg_native_x64_cmake_vs2017_buildonly/tests

**Comment by jtattermusch** [comment]
cl/296531765 should fix the problem

**Comment by jtattermusch** [comment]
Seems fixed by cl/296531765
